src/adaptors/stargate-finance/index.js

```diff
@@ -0,0 +1,219 @@
+
+    return {
+        lpToken,
+        lpTokenSymbol,
+        underlyingLpToken,
+        allocPoint,
+        totalAllocPoint,
+        rewardPerBlock,
+        reserve,
+    };
+}
+
+
+const getPrices = async (chain, addresses) => {
+
+    const prices = (
+        await superagent.post('https://coins.llama.fi/prices').send({
+            coins: addresses.map((address) => `${chain}:${address}`),
+        })
+    ).body.coins;
+
+    const pricesObj = Object.entries(prices).reduce(
+        (acc, [address, price]) => ({
+            ...acc,
+            [address.split(':')[1].toLowerCase()]: price.price,
+        }),
+        {}
+    );
+
+    return pricesObj;
+};
+
+
+const tvl = async (chain, symbol, underlyingLpToken, reserve) => {
+    // total number of coins in pool * coin price
+    let token = underlyingLpToken;
+    if (symbol === 'S*SGETH') {
+        token = CONFIG[chain].ETHER_TOKEN;
+    }
+
+    const price = (await getPrices(chain, [token.toLowerCase()]))[token.toLowerCase()];
+    const reserveUSD = reserve * price;
+
+    return reserveUSD;
+}
+
+
+function calcApy(chain, allocPoint, totalAllocPoint, reward, rewardPrice, reserve) {
+    // pool rewards per year in usd
+    // blocks per year * reward * wieght * price
+
+    // BLOCK_TIME is number of seconds per 1 block settled
+    // get avg block time directly from provider?
+    let BLOCK_TIME = 13.5;
+    if (chain == 'fantom') {
+        BLOCK_TIME = 1;
+    }
+    if (chain == 'optimism') {
+        BLOCK_TIME = 1;
+    }
+    if (chain == 'polygon') {
+        BLOCK_TIME = 2.11;
+    }
+    if (chain == 'avax') {
+        BLOCK_TIME = 2.03;
+    }
+    if (chain == 'bsc') {
+        BLOCK_TIME = 3;
+    }
+
+    const SECONDS_PER_YEAR = 60 * 60 * 24 * 365;
+    const BLOCKS_PER_YEAR = SECONDS_PER_YEAR / BLOCK_TIME;
+
+    const weight = allocPoint / totalAllocPoint;
+    const rewardPerBlock = reward * weight;
+
+    const rewardPerYear = rewardPerBlock / 1e18 * BLOCKS_PER_YEAR;
+
+    const rewardUSD = rewardPerYear * rewardPrice;
+    const apr = rewardUSD / reserve * 100;
+
+    return apr;
+
+}
+
+
+const getApy = async (chain, lpStaking) => {
+
+    let poolsApy = [];
+    const poolLength = parseInt((await sdk.api.abi.call({ abi: abi.poolLength, target: lpStaking, chain })).output);
+
+    let rewardToken = CONFIG[chain].STG_TOKEN;
+    if (chain == 'optimism') {
+        rewardToken = CONFIG[chain].OP_TOKEN;
+    }
+    const rewardPrice = (await getPrices(chain, [rewardToken.toLowerCase()]))[rewardToken.toLowerCase()];
+
+    for (i = 0; i < poolLength; i++) {
+        const pool = await pools(i, chain, lpStaking);
+        const reserveUSD = await tvl(chain, pool.lpTokenSymbol, pool.underlyingLpToken, pool.reserve);
+        const apy = calcApy(chain, pool.allocPoint, pool.totalAllocPoint, pool.rewardPerBlock, rewardPrice, reserveUSD);
+
+        poolsApy.push({
+            pool: pool.lpToken,
+            chain: chain,
+            project: 'stargate',
```

Comment:
project must match the folder name/slug, which i think is stargate-finance?

Reply:
right right ty

Reply:
i believe the original slug/project name was correct but the folder name i changed at the last second... like a dummy because i checked the directory name within defillama-adapters and it is named stargatefinance